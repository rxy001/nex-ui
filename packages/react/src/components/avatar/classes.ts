import { generateUtilityClasses } from '../utils'

export const avatarClasses = generateUtilityClasses('nui-avatar', [
  'root',

  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-xl',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'outlined',

  'img',
])

export const avatarGroupClasses = generateUtilityClasses('nui-avatar-group', [
  'root',
  'surplus',
])
