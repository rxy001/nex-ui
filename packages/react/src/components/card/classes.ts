import { generateUtilityClasses } from '../utils'

export const cardClasses = generateUtilityClasses('nui-card', [
  'root',
  'blurred',
  'hoverable',

  'shadow-xs',
  'shadow-sm',
  'shadow-md',
  'shadow-lg',
  'shadow-xl',

  'radius-none',
  'radius-sm',
  'radius-md',
  'radius-lg',
])

export const cardHeaderClasses = generateUtilityClasses('nui-card-header', [
  'root',
  'content',
  'title',
  'subtitle',
])

export const cardBodyClasses = generateUtilityClasses('nui-card-body', ['root'])

export const cardFooterClasses = generateUtilityClasses('nui-card-footer', [
  'root',
])

export const cardActionAreaClasses = generateUtilityClasses(
  'nui-card-action-area',
  ['root'],
)
