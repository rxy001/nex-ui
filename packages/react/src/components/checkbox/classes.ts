import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const checkboxClasses = generateUtilityClasses('nui-checkbox', [
  // ---------root---------
  'root',

  // "sm" | "md" | "lg" | "full" | "none"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  'indeterminate',
  // ---------root---------

  'input',
  'label',
  'icon',
])

export const checkboxGroupClasses = generateUtilityClasses(
  'nui-checkbox-group',
  [
    'root',
    'label',
    'wrapper',
    'orientation-vertical',
    'orientation-horizontal',
  ],
)
