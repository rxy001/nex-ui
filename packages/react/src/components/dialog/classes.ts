import { generateUtilityClasses } from '../utils'

export const dialogClasses = generateUtilityClasses('nui-dialog', [
  'root',
  'open',
  'backdrop',
])

export const dialogContentClasses = generateUtilityClasses(
  'nui-dialog-content',
  [
    'root',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'full-screen',

    'placement-top',
    'placement-bottom',
    'placement-center',

    'scroll-inside',
    'scroll-outside',

    'paper',
    'close-button',
  ],
)

export const dialogHeaderClasses = generateUtilityClasses('nui-dialog-header', [
  'root',
])

export const dialogFooterClasses = generateUtilityClasses('nui-dialog-footer', [
  'root',
])

export const dialogBodyClasses = generateUtilityClasses('nui-dialog-body', [
  'root',
])
