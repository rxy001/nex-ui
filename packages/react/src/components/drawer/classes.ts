import { generateUtilityClasses } from '../utils'

export const drawerClasses = generateUtilityClasses('nui-drawer', [
  'root',
  'open',
  'backdrop',
])

export const drawerContentClasses = generateUtilityClasses(
  'nui-drawer-content',
  [
    'root',

    'placement-top',
    'placement-right',
    'placement-bottom',
    'placement-left',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'paper',
    'close-button',
  ],
)

export const drawerHeaderClasses = generateUtilityClasses('nui-drawer-header', [
  'root',
])

export const drawerFooterClasses = generateUtilityClasses('nui-drawer-footer', [
  'root',
])

export const drawerBodyClasses = generateUtilityClasses('nui-drawer-body', [
  'root',
])
