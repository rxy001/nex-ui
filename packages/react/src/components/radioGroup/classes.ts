import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const radioClasses = generateUtilityClasses('nui-radio', [
  // ---------root---------
  'root',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  // ---------root---------

  'input',
  'label',
  'dot',
])

export const radioGroupClasses = generateUtilityClasses('nui-radio-group', [
  'root',
  'label',
  'wrapper',
  'orientation-vertical',
  'orientation-horizontal',
])
