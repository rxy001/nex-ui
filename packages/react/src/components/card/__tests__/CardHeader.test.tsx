import {
  renderWithNexUIProvider,
  testComponentStability,
  testRefForwarding,
  testSlotPropsForwarding,
  testClassNamesForwarding,
  testRootClassName,
} from '~/tests/shared'
import { CardHeader } from '../index'
import { cardHeaderClasses } from './classes'

const slots = ['content', 'title', 'subtitle'] as const

describe('CardHeader', () => {
  testComponentStability(<CardHeader title='Title' subtitle='Subtitle' />)

  testRefForwarding(<CardHeader />)

  testRootClassName(<CardHeader />)

  testSlotPropsForwarding(
    <CardHeader title='Title' subtitle='Subtitle' />,
    slots,
    {
      content: { className: 'test-content' },
      subtitle: { className: 'test-subtitle' },
      title: { className: 'test-title' },
    },
    cardHeaderClasses,
  )

  testClassNamesForwarding(
    <CardHeader title='Title' subtitle='Subtitle' />,
    slots,
    {
      content: 'test-content',
      title: 'test-title',
      subtitle: 'test-subtitle',
    },
    cardHeaderClasses,
  )

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<CardHeader />)

    const cardHeaderRoot = container.firstElementChild
    expect(cardHeaderRoot).toHaveClass(cardHeaderClasses.root)

    expect(cardHeaderRoot).toMatchSnapshot()
  })

  it('should render title', () => {
    const { getByText } = renderWithNexUIProvider(<CardHeader title='Title' />)

    const titleElement = getByText('Title')

    expect(titleElement).toBeInTheDocument()
    expect(titleElement).toHaveClass(cardHeaderClasses.title)
  })

  it('should render subtitle', () => {
    const { getByText } = renderWithNexUIProvider(
      <CardHeader subtitle='Subtitle' />,
    )

    const subtitleElement = getByText('Subtitle')

    expect(subtitleElement).toBeInTheDocument()
    expect(subtitleElement).toHaveClass(cardHeaderClasses.subtitle)
  })

  it('should render title and subtitle in the content element', () => {
    const { getByText, queryByClassName } = renderWithNexUIProvider(
      <CardHeader title='Title' subtitle='Subtitle' />,
    )

    const titleElement = getByText('Title')
    const subtitleElement = getByText('Subtitle')

    expect(titleElement).toBeInTheDocument()
    expect(subtitleElement).toBeInTheDocument()
    expect(queryByClassName(cardHeaderClasses.content)).toBeInTheDocument()
  })

  it('should render avatar', () => {
    const { getByText } = renderWithNexUIProvider(
      <CardHeader avatar='Avatar' />,
    )

    expect(getByText('Avatar')).toBeInTheDocument()
  })

  it('should render action', () => {
    const { getByText } = renderWithNexUIProvider(
      <CardHeader action='Action' />,
    )

    expect(getByText('Action')).toBeInTheDocument()
  })

  it('should not render content slot when only avatar and action are provided', () => {
    const { queryByClassName } = renderWithNexUIProvider(
      <CardHeader avatar='Avatar' action='Action' />,
    )

    expect(queryByClassName(cardHeaderClasses.content)).not.toBeInTheDocument()
  })

  it('should have children override action, avatar, title, and subtitle when provided', () => {
    const { queryByText } = renderWithNexUIProvider(
      <CardHeader
        title='Title'
        subtitle='Subtitle'
        avatar='Avatar'
        action='Action'
      >
        Children
      </CardHeader>,
    )

    expect(queryByText('Title')).not.toBeInTheDocument()
    expect(queryByText('Subtitle')).not.toBeInTheDocument()
    expect(queryByText('Avatar')).not.toBeInTheDocument()
    expect(queryByText('Action')).not.toBeInTheDocument()
    expect(queryByText('Children')).toBeInTheDocument()
  })
})
