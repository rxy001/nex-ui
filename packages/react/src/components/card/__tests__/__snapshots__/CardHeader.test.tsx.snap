// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader should render with default props 1`] = `
.emotion-0 {
  padding: var(--nui-spaces-3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--nui-spaces-3);
}

<div
  class="nui-card-header-root emotion-0"
/>
`;
